### PR TITLE
decode: add icmpv4.hdr keyword

### PR DESCRIPTION
**Pull request by jlucovsky**
This PR adds a keyword for the ICMPv4 header -- `icmpv4.hdr`. 

This sticky buffer returns ICMPv4 header bytes; the size of the header is determined by the ICMP message type. Some messages, like router advertisements, are variable lengths.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [4090](https://redmine.openinfosecfoundation.org/issues/4090)

Describe changes:
- New `icmpv4.hdr` sticky buffer

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

suricata-verify-pr: 357
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: src/decode-icmpv4.c [hunk 1]
```diff
@@ -296,6 +313,20 @@ int DecodeICMPV4(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p, const uint8_t
             }
             break;
 
+        case ICMP_ROUTERADVERT: {
+            /* packet points to beginning of icmp header */
```
Comment (catenacyber) [inline src/decode-icmpv4.c:317]: rather to the end actually ?

File: src/decode-icmpv4.c [hunk 2]
```diff
@@ -296,6 +313,20 @@ int DecodeICMPV4(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p, const uint8_t
...
+            ICMPV4RtrAdvert *icmpv4_router_advert = (ICMPV4RtrAdvert *)(pkt + sizeof(ICMPV4Hdr));
```
Comment (catenacyber) [inline src/decode-icmpv4.c:318]: Is not there a check missing like `if (p->payload_len < sizeof(ICMPV4RtrAdvert)) {` ?

Comment (catenacyber) [reply src/decode-icmpv4.c:318]: Ok, no check missing, but misleading comment

Comment (jlucovsky) [reply src/decode-icmpv4.c:318]: Will adjust comment.

**Comment by jlucovsky** [comment]
Continued in #5537

**Pull request closed without merging**